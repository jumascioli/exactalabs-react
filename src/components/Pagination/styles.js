import styled from "styled-components";

export const Wrapper = styled.div`
  align-items: center;
  box-shadow: 2px 2px 4px 0.25px rgba(0, 0, 0, 0.3);
  color: #073b4c;
  display: flex;
  font-size: 16px;
  height: 40px;
  justify-content: space-between;
  padding: 0 20px;
`;

export const Button = styled.button`
  background: transparent;
  border: 0;
  color: #073b4c;
  cursor: pointer;
`;