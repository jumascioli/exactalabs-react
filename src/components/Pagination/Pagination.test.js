import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import Pagination from '.';

describe("<Pagination />", () => {

  it("should render correctly", () => {
    const { container } = render(<Pagination />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it("should render page number", () => {
    render(<Pagination page={1} />);

    const item = screen.getByText(/pÃ¡gina 1/i);

    expect(item).toBeInTheDocument();
  });

  it("should call onPrev onNext", () => {
    const handleNext = jest.fn();
    const handlePrev = jest.fn(); 

    render(<Pagination page={5} onPrev={handlePrev} onNext={handleNext} />);

    const buttonNext = screen.getByRole("button", {
      name: ">",
    });

    userEvent.click(buttonNext);

    expect(handleNext).toBeCalled();
    expect(handlePrev).not.toBeCalled();

    const buttonPrev = screen.getByRole("button", {
      name: "<",
    });

    userEvent.click(buttonPrev);

    expect(handlePrev).toBeCalledTimes(1);
    expect(handleNext).toBeCalledTimes(1);
  });
});