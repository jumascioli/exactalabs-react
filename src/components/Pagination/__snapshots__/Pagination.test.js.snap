// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pagination /> should render correctly 1`] = `
<div
  class="sc-bdfBQB jdGOJV"
>
  <button
    class="sc-gsTEea bgKMIM"
  >
    &lt;
  </button>
  <span>
    PÃ¡gina 
  </span>
  <button
    class="sc-gsTEea bgKMIM"
  >
    &gt;
  </button>
</div>
`;
